src/Symfony/Component/Mailer/Bridge/Mailgun/Transport/MailgunApiTransport.php

```diff
@@ -128,7 +128,17 @@ private function getPayload(Email $email, Envelope $envelope): array
                 continue;
             }
 
-            $payload['h:'.$name] = $header->getBodyAsString();
+            // Check if it is a valid prefix or header name according to Mailgun API
+            $prefix = substr($name, 0, 2);
+            if (\in_array($prefix, ['h:', 't:', 'o:', 'v:']) || \in_array($name, ['recipient-variables', 'template', 'amp-html'])) {
+                $headerName = $name;
+            } else {
+                // fallback to prefix with "h:" to not break BC
+                $headerName = 'h:'.$name;
+                @trigger_error(sprintf('Not prefixing the Mailgun header name with "h:" is deprecated since Symfony  5.1. Use header name "%s" instead.', $headerName), E_USER_DEPRECATED);
```

Comment:
This seems weird to me. The `h:` prefix is a mailgun specific feature. Requiring the Email object to set it means writing Mailgun-specific code in a place that may not even be aware of which transport is used.

Reply:
@Nyholm I suggest removing the deprecation warning